### PR TITLE
User profile, user stories 19-22

### PR DESCRIPTION
**Pull request by jdcarey128**
This PR: 
- Allows users to see and edit their profile including their email and password 
- Tests in registered_user_profile_spec.rb
- Adds several partials for user forms (In future, will want to refactor to be more DRY)
- Adds format_date helper method  in helpers directory 
- Defines edit and update actions in users_controller
- Re-writes user password validation to allow for users to update profile without password
- All tests passing

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+
+  def format_date(input)
```
Comment (smcunning) [inline app/helpers/application_helper.rb:3]: Very cool, looking forward to using this in some of the views I'm working on!

File: app/views/users/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<%= render partial: @render, locals: {
```
Comment (smcunning) [inline app/views/users/edit.html.erb:1]: Nice! Love to see setting up these partials.

File: app/views/users/_edit_profile.html.erb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  <%= label_tag :zip %>
+  <%= text_field_tag :zip, zip_value %>
+
+  <%= label_tag :email %>
+  <%= text_field_tag :email, email_value %>
+
+  <%= hidden_field_tag :old_email, original_email_value %>
+
+  <%= yield %>
```
Comment (smcunning) [inline app/views/users/_edit_profile.html.erb:22]: Would love to learn more about what this yield is doing

**Review by smcunning — APPROVED** [review]
Well done on these! Ran RSpec locally and all tests pass. Good to merge.

File: app/views/users/_edit_profile.html.erb (hunk 3 above)
Comment (katemorris) [reply app/views/users/_edit_profile.html.erb:22]: @jdcarey128 Did you get the yield working in the partial?

**Review by katemorris — CHANGES_REQUESTED** [review]
Before I approve, just need to make sure that yield is working as expected. If it is, that is awesome. If not, please remove and let me know, I'll change to approve.

**Pull request merged**